### ISSUE TITLE
[pickers] Can't use the lab with React 18 due to pickers version

### ISSUE THREAD

**Issue by PCOffline** [mui/mui-x#4932]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

As of right now, working with React 18 causes a peer dependency conflict due to date-pickers relying on React 17

**Comment by flaviendelangle** [comment]
Hi,

Which version of the pickers are your using ?
Our `package.json` accepts React 18 since `v5.0.0-alpha.1`

**Comment by PCOffline** [comment]
It's a dependency of the latest version of @mui/lab apparently, which uses 5.0.0-alpha.0

**Comment by flaviendelangle** [comment]
Oh right... I will update the lab deps to the `alpha.1` to unlock you

**Comment by mvestergaard** [comment]
A slightly related issue I've just encountered:
`@mui/lab` uses `alpha.0`
`@mui/x-date-pickers-pro` uses `alpha.3`

I end up having both of them, I `import { LocalizationProvider } from '@mui/x-date-pickers'` but any `DateRangePicker` component from `@mui/x-date-pickers-pro` then fails claiming the context is missing, but it's because it's using a context from a different version.

TL;DR. Please align versions between packages, or at least make them accept a later version.

**Comment by mvestergaard** [comment]
> Why do you use the `LocalizationProvider` from `@mui/lab` ?

I don't, but we use other things from `@mui/lab` so get the dependency.
These are the MUI packages we use:
```
    "@mui/lab": "^5.0.0-alpha.82",
    "@mui/material": "^5.8.0",
    "@mui/system": "^5.8.0",
    "@mui/x-data-grid-pro": "^5.11.0",
    "@mui/x-date-pickers-pro": "^5.0.0-alpha.3",
```
With the combination of these versions I end up with:
`node_modules/@mui/x-date-pickers` = `5.0.0-alpha.0`
`node_modules/@mui/x-date-pickers-pro/node_modules/@mui/x-date-pickers` = `5.0.0-alpha.3`

Using yarn if that's not clear

**Comment by flaviendelangle** [comment]
My bad

Does `yarn deduplicate` solves your problem ?

**Comment by mvestergaard** [comment]
> Does `yarn deduplicate` solves your problem ?

I think that's a yarn v1 thing, we use yarn v2, should do deduplication by default

**Comment by mvestergaard** [comment]
Tried `yarn dedupe @mui/x-date-pickers --strategy highest` but didn't help.

I think it's because `@mui/lab` depends on `alpha.0` specifically, with no option to upgrade, so it doesn't have a choice but to have both.

**Comment by flaviendelangle** [comment]
@oliviertassinari / @cherniavskii do you have a solution ?

**Comment by oliviertassinari** [comment]
From what I understand @mvestergaard imports from `@mui/x-date-pickers`:

```jsx
import { LocalizationProvider } from '@mui/x-date-pickers'
```

but doesn't have it declared in its dependencies:

```json
    "@mui/lab": "^5.0.0-alpha.82",
    "@mui/material": "^5.8.0",
    "@mui/system": "^5.8.0",
    "@mui/x-data-grid-pro": "^5.11.0",
    "@mui/x-date-pickers-pro": "^5.0.0-alpha.3",
```

so it doesn't work. This looks like the expected outcome.

**Comment by mvestergaard** [comment]
@oliviertassinari You're right that adding `@mui/x-date-pickers` explicitly avoids the issues.
I don't know if I'm naive to expect to only need to add the pro one, and let that dictate the version.
